src/components/atoms/Input/Input.test.js

```diff
@@ -0,0 +1,318 @@
+import React from 'react'
+import { fireEvent } from '@testing-library/react'
+import 'jest-styled-components'
+
+import { Input } from './Input'
+import { renderWithProviders } from '../../../context/__mocks__/mockContext'
+
+describe('Input components', () => {
+  let style
+  let name
+  let type
+  let icon
+  let label
+  let autoComplete
+  let autoFocus
+
+  let component
+  let props
+
+  beforeEach(() => {
+    style = { marginBottom: 0 }
+    name = 'input'
+    icon = 'shop'
+    label = 'Label'
+    autoComplete = 'off'
+    autoFocus = true
+  })
+
+  describe('Text', () => {
+    beforeEach(() => {
+      type = 'text'
+      props = {
+        style, name, type, icon, label, autoComplete, autoFocus
+      }
+      component = renderWithProviders(<Input
+        {...props}
+      />)
+    })
+
+    it('should match snapshot', () => {
+      const { asFragment } = component
+      expect(asFragment()).toMatchSnapshot()
+    })
+
+    it('should print correct label', () => {
+      const { container } = component
+      const span = container.querySelector('.input-label')
+
+      expect(span.textContent).toBe(label)
+    })
+
+    it('should allow letters to be inputted', () => {
+      const { container } = component
+      const elem = container.querySelector('#input')
+
+      fireEvent.change(elem, { target: { value: 'abc23!@' } })
+      expect(elem.value).toBe('abc23!@')
+    })
+
+    it('should allow letters to be deleted', () => {
+      const { container } = component
+      const elem = container.querySelector('#input')
+
+      fireEvent.change(elem, { target: { value: 'abc23!@' } })
+      fireEvent.change(elem, { target: { value: 'abc2' } })
+      expect(elem.value).toBe('abc2')
+    })
+  })
```

Comment:
Since the `allow letters to be deleted`  test also test for `letters to be inputted` just merge them together:

```suggestion
    it('should allow letters to be inputted and deleted', () => {
      const { container } = component
      const elem = container.querySelector('#input')

      // add
      fireEvent.change(elem, { target: { value: 'abc23!@' } })
      expect(elem.value).toBe('abc23!@')
      
      // delete
      fireEvent.change(elem, { target: { value: 'abc2' } })
      expect(elem.value).toBe('abc2')
    })
  })
```

Reply:
as we were previously informed to keep one `expect()` assertion per `it()` block, I had kept insertion and deletion separate